### ISSUE TITLE
serviceName and serviceDescription are not xml-escaped

### ISSUE THREAD

**Issue by rslinckx** [onelogin/python3-saml#67]
When building metadata XML from config dict, the keys serviceName and serviceDescription are included using %s in XML string concatenation.

If they contain for example & they will generate a malformed XML document and libxml2 will error out.

They should be xml encoded before being included in the resulting xml document.

This might affect also other values ?

Note: This can introduce a breaking change if people noticed that, and provide an xml-encoded value in serviceName and serviceDescription, resulting in a double encoding.

**Comment by pitbulk** [comment]
mmm so what is your suggestion? may we just xml-scape the value and put a want on next release? do you have the PR?

**Comment by rslinckx** [comment]
I have no PR, i'd suggest xml-escaping and warning users about the behavior change.

In my code i just import html; html.escape(service_name), but I guess it would be best to use libxml2 to build the xml document instead of concatenating strings?

**Comment by kstrauser** [comment]
I strongly agree with @rslinckx. String concatenation is a significant security risk unless 100% of input data is completely user-generated.
